test/cql-pytest/test_native_functions.py

```diff
@@ -121,3 +122,42 @@ def test_mintimeuuid_extreme_from_totimestamp(cql, table1):
         cql.execute(f"SELECT * FROM {table1} WHERE p={p} and u < mintimeuuid(totimestamp(123)) ALLOW FILTERING")
     except:
         pass
+
+# According to the documentation, the toTimestamp() function can take either
+# a "timeuuid" or a "date" value and convert it to a "timestamp" type
+# (64-bit signed integer representing number of milliseconds since the UNIX
+# epoch). Let's test these conversions, and their error cases (especially,
+# the range dates covered by the different types isn't identical).
+# In test_type_date.py we have test coverage on the different ways a "date"
+# value can initialized, so we don't need to cover all of these here.
+def test_totimestamp_date(cql, table1):
+    p = unique_key_int()
+    # The date 2**31 is the day of the epoch, so 2**31+1 is one day later,
+    # Midnight January 2nd, 1970:
+    cql.execute(f"INSERT INTO {table1} (p, d) VALUES ({p}, {2**31+1})")
+    assert [(datetime(1970,1,2,0,0),)] == list(cql.execute(f"SELECT totimestamp(d) FROM {table1} WHERE p={p}"))
+    # The day 2**31-1 is one day before the epoch, and this (negative
+    # timestamps) is allowed:
+    cql.execute(f"INSERT INTO {table1} (p, d) VALUES ({p}, {2**31-1})")
+    assert [(datetime(1969,12,31,0,0),)] == list(cql.execute(f"SELECT totimestamp(d) FROM {table1} WHERE p={p}"))
+
+# Same as above test test_totimestame_date(), but use extreme dates
+# millions of years in the past. These tests cannot be run with the
```

Comment:
If this test cannot be run with the current Python driver because of its bugs, you could introduce the same test, but using a testing framework where it'd work. Currently you're adding a testcase that may never run.